plugins/woocommerce/includes/tracks/events/class-wc-orders-tracking.php

```diff
@@ -19,53 +19,49 @@ class WC_Orders_Tracking {
 	 */
 	public function init() {
 		add_action( 'woocommerce_order_status_changed', array( $this, 'track_order_status_change' ), 10, 3 );
-		add_action( 'load-edit.php', array( $this, 'track_orders_view' ), 10 );
-		add_action( 'pre_post_update', array( $this, 'track_created_date_change' ), 10 );
 		// WC_Meta_Box_Order_Actions::save() hooks in at priority 50.
 		add_action( 'woocommerce_process_shop_order_meta', array( $this, 'track_order_action' ), 51 );
+
+		add_action( 'load-edit.php', array( $this, 'track_orders_view' ), 10 );
+		add_action( 'load-woocommerce_page_wc-orders', array( $this, 'track_orders_view' ), 10 ); // HPOS.
+
 		add_action( 'load-post-new.php', array( $this, 'track_add_order_from_edit' ), 10 );
-		add_filter( 'woocommerce_shop_order_search_results', array( $this, 'track_order_search' ), 10, 3 );
+		add_action( 'load-woocommerce_page_wc-orders', array( $this, 'track_add_order_from_edit' ), 10 ); // HPOS.
+
+		add_action( 'woocommerce_process_shop_order_meta', array( $this, 'track_created_date_change' ), 10 );
+
+		add_action( 'load-edit.php', array( $this, 'track_order_search' ) );
+		add_action( 'load-woocommerce_page_wc-orders', array( $this, 'track_order_search' ) ); // HPOS.
+
 		add_action( 'admin_enqueue_scripts', array( $this, 'possibly_add_order_tracking_scripts' ) );
 	}
 
 	/**
 	 * Send a track event when on the Order Listing page, and search results are being displayed.
-	 *
-	 * @param array  $order_ids Array of order_ids that are matches for the search.
-	 * @param string $term The string that was used in the search.
-	 * @param array  $search_fields Fields that were used in the original search.
 	 */
-	public function track_order_search( $order_ids, $term, $search_fields ) {
-		// Since `woocommerce_shop_order_search_results` can run in the front-end context, exit if get_current_screen isn't defined.
-		if ( ! function_exists( 'get_current_screen' ) ) {
-			return $order_ids;
-		}
-
-		$screen = get_current_screen();
-
-		// We only want to record this track when the filter is executed on the order listing page.
-		if ( 'edit-shop_order' === $screen->id ) {
-			// we are on the order listing page, and query results are being shown.
-			WC_Tracks::record_event( 'orders_view_search' );
+	public function track_order_search() {
+		if ( ! OrderUtil::is_order_list_table_screen() || empty( $_REQUEST['s'] ) ) { // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+			return;
 		}
 
-		return $order_ids;
+		WC_Tracks::record_event( 'orders_view_search' );
 	}
```

Comment:
This pretty significantly changes both the method signature and its return value. I don't get any errors related to this when testing, but are we sure this isn't going to break anything?

One alternative approach here might be to add a new method with a slightly different name that does exactly what you're doing here, and then deprecate the previous method (and update all the usages of the old method in WC Core). That way the deprecated method can redirect to the new method, but won't cause problems if something outside of WC Core happens to be using it, for whatever reason.

Reply:
I didn't really think much about this as there weren't any other usages in core and it seems to me this is an internal method, only public because it needed to hook onto a WP filter. In any case, I believe your suggestion makes sense and I've deprecated this method (just in case) and used a different name for the new one.

See 4e44411d41cb8231cb73fd99f8d153c189911e26.